src/Channel.php

```diff
@@ -97,6 +105,16 @@ public function events(): Events
         return $this->events;
     }
 
+    private function assertOpen(): void {
+        if ($this->state !== self::STATE_OPEN) {
```

Comment:
It feels more robust to me to check if a channel is open before sending a command instead of letting the server complain about it and handling the error async